photon-core/src/main/java/org/photonvision/vision/camera/CameraInfo.java

```diff
@@ -19,105 +19,204 @@
      */
-    public boolean getIsV4lCsiCamera() {
-        return (Arrays.stream(otherPaths).anyMatch(it -> it.contains("csi-video"))
-                || getBaseName().equals("unicam"));
-    }
+    String path();
 
     /**
      * @return The base name of the camera aka the name as just ascii.
      */
-    public String getBaseName() {
-        return name.replaceAll("[^\\x00-\\x7F]", "");
-    }
+    String name();
 
     /**
      * @return Returns a human readable name
      */
-    public String getHumanReadableName() {
-        return getBaseName().replaceAll(" ", "_");
+    default String humanReadableName() {
+        return name().replaceAll(" ", "_");
     }
 
     /**
-     * Get a unique descriptor of the USB port this camera is attached to. EG
-     * "/dev/v4l/by-path/platform-fc800000.usb-usb-0:1.3:1.0-video-index0"
+     * If the camera is a USB camera this method returns
+     * a unique descriptor of the USB port this camera is attached to. EG
+     * "/dev/v4l/by-path/platform-fc800000.usb-usb-0:1.3:1.0-video-index0".
+     * If the camera is a CSI camera this method returns the path of the camera.
      *
-     * @return
+     * @return The unique path of the camera
      */
-    public Optional<String> getUSBPath() {
-        return Arrays.stream(otherPaths).filter(path -> path.contains("/by-path/")).findFirst();
+    String uniquePath();
+
+    String[] otherPaths();
+
+    CameraType type();
+
+    public static final class PVUsbCameraInfo extends UsbCameraInfo implements CameraInfo {
+
+        private PVUsbCameraInfo(
+                int dev, String path, String name, String[] otherPaths, int vendorId, int productId) {
+            super(dev, path, name, otherPaths, vendorId, productId);
+        }
+
+        private PVUsbCameraInfo(UsbCameraInfo info) {
+            super(info.dev, info.path, info.name, info.otherPaths, info.vendorId, info.productId);
+        }
+
+        @Override
+        public String path() {
+            return super.path;
+        }
+
+        @Override
+        public String name() {
+            return super.name.replaceAll("[^\\x00-\\x7F]", "");
+        }
+
+        @Override
+        public String uniquePath() {
+            return Arrays.stream(super.otherPaths).filter(path -> path.contains("/by-path/")).findFirst()
+                    .orElse(path());
+        }
+
+        @Override
+        public String[] otherPaths() {
+            return super.otherPaths;
+        }
+
+        @Override
+        public CameraType type() {
+            return CameraType.UsbCamera;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (obj == null) return false;
+            if (getClass() != obj.getClass()) return false;
+            PVUsbCameraInfo other = (PVUsbCameraInfo) obj;
+
+            // Windows device number is not significant. See
```

Comment:
This is kinda-sorta true on Linux, too. /dev/videoN isn't consistent across reboots or hot-plugs.

Reply:
This was just copy pasted over from old code, ill look more into it.